lib/boombox.ex

```diff
@@ -286,6 +344,28 @@ defmodule Boombox do
     end
   end
 
+  @spec await_source_ready() :: pid()
+  defp await_source_ready() do
+    receive do
+      {:boombox_ex_stream_source, source} -> source
+    end
+  end
+
+  @spec await_sink_ready() :: pid()
+  defp await_sink_ready() do
+    receive do
+      {:boombox_ex_stream_sink, sink} -> sink
+    end
+  end
+
+  @spec await_external_resource_ready() :: :ok
```

Comment:
[NIT] `external_resource` might be slightly vague if somebody is not familiar with that concept, but I am not sure I can find better name - how about we just add some comment here saying that we wait for e.g. UDP socket to start listening?